### PR TITLE
[ITensorMPS] Fix some lingering namespace issues

### PR DESCRIPTION
**Pull request by emstoudenmire**
Fixes of test errors in `test/ITensorMPS/Ops/` and the `ITensorGaussianMPS` package.

Also attempting to fix the error seen when running the `ITensorGPU` tests.

### CODE REVIEW

**Comment by emstoudenmire** [comment]
[test ITensors mps]

**Comment by emstoudenmire** [comment]
[test ITensorGaussianMPS]

**Comment by emstoudenmire** [comment]
So the test failure here for the MPS tests is strange: apparently `allequal` was not defined in Base until Julia 1.8. But it is used in some `@assert`'s in the file `abstractprojmpo/projmposum.jl`. So I'm surprised these tests were ever passing. 

The question is whether we should:
1) replace the use of `allequal` with something else there or
2) raise the minimum Julia version to 1.8

I'd be fine with either but I think for now better to just replace `allequal` if we are just using it in a small number of places.

**Comment by emstoudenmire** [comment]
[test ITensors mps]

**Comment by emstoudenmire** [comment]
[test ITensorGaussianMPS]

**Comment by emstoudenmire** [comment]
I went ahead and replaced allequal but let's discuss which choice would be best.

**Comment by mtfishman** [comment]
> So the test failure here for the MPS tests is strange: apparently `allequal` was not defined in Base until Julia 1.8. But it is used in some `@assert`'s in the file `abstractprojmpo/projmposum.jl`. So I'm surprised these tests were ever passing.
> 
> The question is whether we should:
> 
>     1. replace the use of `allequal` with something else there or
> 
>     2. raise the minimum Julia version to 1.8
> 
> 
> I'd be fine with either but I think for now better to just replace `allequal` if we are just using it in a small number of places.

I've been using the [Compat.jl](https://github.com/JuliaLang/Compat.jl) package to address issues like that. `Compat.jl` makes some subset of newer Julia functionality (like `allequal`) available in older versions of Julia. Probably it was working before because we load `Compat` in `ITensors.jl`, but it isn't being loaded in the `ITensorMPS` module now. So we could add `using Compat: allequal` at the top of the `abstractprojmpo/projmposum.jl` file.

**Comment by mtfishman** [comment]
> [test ITensors mps]

We should change the trigger comment for this workflow to `[test ITensorMPS]`, it just requires changing this line: https://github.com/ITensor/ITensors.jl/blob/v0.3.57/.github/workflows/comment_trigger_test_itensors_mps.yml#L12.

File: ITensorGaussianMPS/src/gmps.jl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import Base: sortperm, size, length, eltype, conj, transpose, copy, *
-using ITensors: alias
+using ITensors: alias, ITensorMPS
```
Comment (mtfishman) [inline ITensorGaussianMPS/src/gmps.jl:2]:
```suggestion
using ITensors.ITensorMPS: ITensorMPS
using ITensors: alias
```
related to my comment here: https://github.com/mtfishman/ITensorChemistry.jl/pull/9/files#r1537556024

File: test/ITensorMPS/Ops/test_ops_mpo.jl [hunk 2]
```diff
@@ -1,6 +1,7 @@
 using Test
 using ITensors
 using ITensors.Ops
+using ITensors: ITensorMPS
```
Comment (mtfishman) [inline test/ITensorMPS/Ops/test_ops_mpo.jl:4]:
```suggestion
using ITensors.ITensorMPS: ITensorMPS
```

File: src/ITensorMPS/dmrg.jl [hunk 3]
```diff
@@ -249,12 +249,11 @@ function dmrg(
...
         ## Adapt is only called when using CUDA backend. CPU will work as implemented previously.
         ## TODO this might be the only place we really need iscu if its not fixed.
-        phi::ITensor = if NDTensors.iscu(phi) && NDTensors.iscu(vecs[1])
-          adapt(set_eltype(unwrap_array_type(phi), eltype(vecs[1])), vecs[1])
+        if NDTensors.iscu(phi) && NDTensors.iscu(vecs[1])
+          phi = adapt(ITensors.set_eltype(unwrap_array_type(phi), eltype(vecs[1])), vecs[1])
         else
-          vecs[1]
+          phi = vecs[1]
         end
```
[the remark this answers is not in the document]
Comment (emstoudenmire) [reply src/ITensorMPS/dmrg.jl:256]: That makes sense – I'll put it back

File: .github/workflows/comment_trigger_test_itensors_mps.yml [hunk 4]
```diff
@@ -9,7 +9,7 @@ on:
 jobs:
   test:
     name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ matrix.threads }} thread(s)
-    if: github.event.issue.pull_request && contains(github.event.comment.body, '[test ITensors mps]')
+    if: github.event.issue.pull_request && contains(github.event.comment.body, '[test ITensorMPS]')
```
[the remark this answers is not in the document]
Comment (emstoudenmire) [reply .github/workflows/comment_trigger_test_itensors_mps.yml:12]: Thanks – make sense

**Comment by emstoudenmire** [comment]
[test ITensorGaussianMPS]

**Pull request merged**